### PR TITLE
[KOGITO-3481] - Review runtime specific configuration by operator

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Many thanks for submiting your Pull Request :heart:! 

https://issues.redhat.com/browse/KOGITO-3481

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: pkg/controller/kogitojobsservice/kogitojobsservice_controller.go [hunk 1]
```diff
@@ -137,27 +134,9 @@ func (r *ReconcileKogitoJobsService) Reconcile(request reconcile.Request) (resul
...
-
-func onDeploymentCreate(cli *kogitocli.Client, deployment *appsv1.Deployment, service appv1alpha1.KogitoService) error {
-	jobService := service.(*appv1alpha1.KogitoJobsService)
-	if jobService.Spec.BackOffRetryMillis <= 0 {
-		jobService.Spec.BackOffRetryMillis = backOffRetryDefaultValue
-	}
-	framework.SetEnvVar(backOffRetryEnvKey, strconv.FormatInt(jobService.Spec.BackOffRetryMillis, 10), &deployment.Spec.Template.Spec.Containers[0])
-	if jobService.Spec.MaxIntervalLimitToRetryMillis <= 0 {
-		jobService.Spec.MaxIntervalLimitToRetryMillis = maxIntervalLimitRetryDefaultValue
-	}
```
Comment (radtriste) [inline pkg/controller/kogitojobsservice/kogitojobsservice_controller.go:159]: we don't define default values for those anymore ?

Comment (Kaitou786) [reply pkg/controller/kogitojobsservice/kogitojobsservice_controller.go:159]: I think the default values are already defined at the runtime level

**Review by Kaitou786 — APPROVED** [review]

File: pkg/controller/kogitojobsservice/kogitojobsservice_controller.go (hunk 1 above)
Comment (vaibhavjainwiz) [reply pkg/controller/kogitojobsservice/kogitojobsservice_controller.go:159]: @radtriste yes, now operator don't define any values to service configurations. 
Idea with this story is that, defining application properties should be a user resonposiblity. Operator just receive these configuration in key/value pair under `Spec.Config` and append them in `application.properties` into ConfigMap.

Comment (radtriste) [reply pkg/controller/kogitojobsservice/kogitojobsservice_controller.go:159]: good for me then :)

**Review by radtriste — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
Waiting for QE verification

/jenkins test

File: pkg/apis/app/v1alpha1/kogitoservices_types.go [hunk 2]
```diff
@@ -193,6 +194,13 @@ type KogitoServiceSpec struct {
 	// Create Service monitor instance to connect with Monitoring service
 	// +optional
 	Monitoring Monitoring `json:"monitoring,omitempty"`
+
+	// +optional
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Configs"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	// Application properties that will be set to the service. For example 'MY_VAR: my_value'.
```
Comment (sutaakar) [inline pkg/apis/app/v1alpha1/kogitoservices_types.go:202]: Shouldn't be the example rather `my_var=my_value`?

Comment (ricardozanini) [reply pkg/apis/app/v1alpha1/kogitoservices_types.go:202]: The properties are set directly to the yaml file like:

```yaml
config:
    my.application.key: value
```

So it makes sense.

Comment (sutaakar) [reply pkg/apis/app/v1alpha1/kogitoservices_types.go:202]: right, I see

**Review by sutaakar — APPROVED** [review]
Looks good to me.

**Pull request merged**